src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -225,12 +226,31 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 properties = properties.SetItem("subscriptionId", new TypeProperty("subscriptionId", LanguageConstants.String, TypePropertyFlags.DeployTimeConstant));
             }
 
+            var functions = GetTypeBicepFunctions(typeReference);
+
             return new ObjectType(
                 objectType.Name,
                 objectType.ValidationFlags,
                 isExistingResource ? ConvertToReadOnly(properties.Values) : properties.Values,
                 objectType.AdditionalPropertiesType,
-                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags);
+                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags,
+                functions);
+        }
+
+        private static IEnumerable<FunctionOverload> GetTypeBicepFunctions(ResourceTypeReference resourceType)
```

Comment:
>GetTypeBicepFunctions [](http://example.com/codeflow?start=53&length=21)

Can you rename to `GetBuiltInMethods` or `GetBicepMethods`?

Reply:
Renamed to GetBicepMethods